tools/SourceKit/lib/SwiftLang/SwiftSourceDocInfo.cpp

```diff
@@ -755,7 +755,7 @@ static bool passCursorInfoForDecl(const ValueDecl *VD,
         PO.SkipIntroducerKeywords = true;
         PO.ArgAndParamPrinting = PrintOptions::ArgAndParamPrintingMode::ArgumentOnly;
         XMLEscapingPrinter Printer(RelDeclsStream);
-        if (BaseType) {
+        if (BaseType && RelatedDecl->getDeclContext() == BaseType->getAnyNominal()) {
```

Comment:
UnqualifiedLookup returns the BaseDecl as part of each lookup result. If it's a self ParamDecl, set your BaseType to something. Otherwise, don't set it.